### ISSUE TITLE
Inviting Members to Sealed Topic

### ISSUE THREAD

**Issue by tokenwizard** [balzack/databag#126]
I have created a Public Topic and a Sealed Topic on my server. 
On the Public Topic, there is a button on the right to manage Members for that Topic:
![image](https://github.com/user-attachments/assets/69477149-01e5-4e13-bb29-80936b8377eb)

I cannot find any such option for the Sealed topic:
![image](https://github.com/user-attachments/assets/348d2a01-b649-4547-9361-55f7be27a391)

Once you create Sealed Topics, how are you supposed to add/invite members to it?

**Comment by balzack** [comment]
Currently all members to a sealed topic must be set at topic creation time.

I am working on a refactor, which will include the ability to add members after topic creation time, but this is still a couple months away. Sorry for the inconvenience.

**Comment by tokenwizard** [comment]
OK, sounds good.
Not a big deal as I am just sorta "beta testing" this with a couple friends.

Sent from Proton Mail Android

-------- Original Message --------

**Comment by tokenwizard** [comment]
I would be glad to put together some bullet-point style pointers for new users. I am already working on an onboarding quick-start guide to send to less tech-savvy friends and family to make their registration and app setup process simpler. 

But there are still some quirks I need to figure out myself before I can put together fairly complete documentation. 

Related to Members and Topics:

It seems when a new member joins, you have to first Connect with them via Contacts before you can add them to an Unsealed Topic. Let's say there is a conversation going between three connected people in a Topic, and you have a new member join. If only ONE of the three that are already connected Connects with the new member and then that member gets added to the Topic, will that new member be able to see the topic posts from the other two who they are not connected with directly?

**Comment by balzack** [comment]
yes, an on-boarding doc would be great!

I am not sure if I am understanding your question, so correct me if I misunderstand.

You create a topic with contact-A and contact-B with whom you are connected:
1. contact-A and contact-B have not saved each others contacts (so they don't appear in their contact list): they will see each others messages, but the message header will be from 'unknown' contact
2. only contact-A has saved contact-B: contact-A will see the message header of contact-B, but contact-B will see the message header of 'unknown' for contact-A's posts.
3. contact-A and contact-B have saved each others contacts (may also be connected, but not required): they will see each others messages with the contact's message header

**Comment by tokenwizard** [comment]
Kinda. Maybe? 
Let's say James and Bill are connected to each other and are part of Topic called Cars. 
Those two can chat back and forth and see each other's messages and their names. 

Today, Peter signed up for Databag and James Connects with Peter but Bill does not. 

Now if Peter gets added to the Topic called Cars, will Peter see the messages posted by Bill, even if they are not connected to Peter?

**Comment by balzack** [comment]
Thanks for clarifying....yes, Peter will see the messages posted by Bill.

some additional detail since a contact can be saved even though not connected:
If Peter has not saved Bill's contact, the message header on messages from Peter will appear from an 'unknown' contact.
If Peter has saved Bill's contact (even though not connected), the message header in the thread will appear from contact 'Bill'

**Comment by tokenwizard** [comment]
Ok, I understand your explanation, but could you please clarify the difference between Saving a contact and Connecting one? Is Saving more like just recognizing their existing where Connecting means you are "friends" with them? Are there other variances in behavior for one vs the other besides them just showing as unknown? 

It seems we will want the Guide to point out these intricacies and what to do if someone is in a Topic and sees Unknown Users. 

What are the benefits of having both options? On other platforms I've tested, anyone who is part of a server can see all other members of the server, without the added steps of Saving or Connecting with them.

**Comment by balzack** [comment]
Yes, your understanding is correct, it's just recognizing their existence.

Saving a contact means you just add them to you contact list. You can do this by browsing the registry and 'saving' the contact, or if they try and connect to you, you can choose to save the contact, but not connect. 

In practice, I don't think a 'saved' contact without wanting a connection has much value, but internally it's the first step in the connection process, and I think I will have more use for that state as the project matures.

**Comment by tokenwizard** [comment]
Having some odd issues with this at the moment. I have one friend who joined, I Added him as a Connection, and all is good. We can post back and forth in the Public Topic I have. But my brother joined and I did add him as a Contact as a Connection as well. He has since updated his Profile Pic and I cannot see it in my Contact List, but if I click Add to see all the server contacts, I see the pic there. When I click him there, it ways we are connected. 

Also, he cannot see any of the posts in the Public topic that I added him to. 

I feel like there is just a learning curve I am slamming against and once I figure things out, I can document the process to make it easier for everyone else who is a newcomer.

**Comment by balzack** [comment]
It seems like there is some network communication error. 

In the admin config, did you set a value for federated host name?
Is your brother accessing the server via an internal network IP or a public host name?
Is the way your brother connects to his account any different from the way your friend does? (home area network vs public)

**Comment by tokenwizard** [comment]
Yes, I set my server host in the settings for federation, but the two other users here are on my server, so I wouldn't think federation would be involved?

Both users are accessing via the invite I sent them from the admin panel, via the hostname, not an IP.
The only difference I can think of is my brother is in Germany and the rest of us are in the US. He is going to try accessing via VPN to see if it makes a difference.

-------- Original Message --------

**Comment by balzack** [comment]
Are they both using the mobile app, or both using the browser app?

The federated host name should not be necessary. Internally it indicates whether a contact is assumed to reside on the same server or not. You should be allowed to set it at anytime, but perhaps there is some bug if it was set in between when your friend created the account and when your brother did.

I'm trying to think of any configuration changes that could have occurred between the account creation/access. Residing in Germany should not be an issue. I will try and reproduce in my test environment. Thanks for bringing this to my attention!

**Comment by tokenwizard** [comment]
I believe both of them registered using the web app and then installed the Android app and signed in. The federation address was set when I installed the server, before I sent invites to either of them.

Let me know what I can to to help troubleshoot.

-------- Original Message --------

**Comment by balzack** [comment]
Would your brother know how to look at the debug console of the browser? If so could he login with the browser and see if any error messages appear? Thank you for your help.

**Comment by tokenwizard** [comment]
I will ask.

-------- Original Message --------

**Comment by tokenwizard** [comment]
Ok, I let him know what we needed. Not sure when he will have time to get around to it.

**Comment by tokenwizard** [comment]
If you want, I can DM you my account if you want to connect and do any testing that way, too.

**Comment by balzack** [comment]
Yes, if you don't mind. Could you connect to my account on the demo server, and message me over databag?
username: abeam
server: databag.coredb.org

**Comment by balzack** [comment]
The screen only returns the first 32 results, you need to click the filter button on the right, and add the username....sorry, the messaging for that feature is poor.
